lib/derivation.ml

```diff
@@ -15,6 +15,10 @@ let rec derive context = function
      | None ->
         Error ("unknown type variable: " ^ v)
      end
+  | Struct exprs ->
+     result_map_m ~f:(derive context) exprs
+     >>| List.unzip
+     >>| fun (tys, cs) -> (TyStruct tys, Conj cs)
```

Comment:
I would really appreciate it if you would add some simple unit tests for STRUCT derivation.

Reply:
A test introduced, thanks!